FireRoad/CourseBrowserViewController.swift

```diff
@@ -393,7 +396,34 @@ class CourseBrowserViewController: UIViewController, UISearchBarDelegate, UITabl
                             newAggregatedSearchResults[course] = relevance + log(max(user.userRelevance(for: course), 2.0))
                         }
                     }
-                    let sortedResults = newAggregatedSearchResults.sorted(by: { $0.1 > $1.1 }).map { $0.0 }
+                    func sortingFunction(course1: (key: Course, value: Float), course2: (key: Course, value: Float)) -> Bool {
```

Comment:
I feel like 3-time duplicated code is a bit too much. Can you define a helper class (e.g. `CourseSorting` or `CourseSortHelper`) that declares these sorting functions as methods? Try to condense them into one implementation, and offer two alternative method signatures for the two different input parameter types (`Course` and `(Course, Float)`). 

Since the sort option is part of the input to the function, you probably want to give the helper object the sort option as a property:

```
class CourseSortHelper {
    var sortType: SortOption
    init(sortType: SortOption) {
        // ...
    }
    func sortingFunction(...) {
        // Use sortType to figure out what ordering to use
    }
}
```